keras/layers/rnn/lstm.py

```diff
@@ -1256,21 +1253,6 @@ def gpu_lstm_with_fallback(
         return_sequences,
     ):
         """Use cuDNN kernel when mask is none or strictly right padded."""
-        if mask is None:
```

Comment:
A potential performance issue for cuDNN users here is that there is a `tf.cond` op inserted in the graph (which did not exist before this change). This can create some (albeit minor) overhead.

Please do the following:

- Use `tf.__internal__.smart_cond.smart_cond` instead of `tf.cond`
- In `is_cuddn_supported`, use True/False instead of `tf.constant(True/False)`